src/bin/nydusd/main.rs

```diff
@@ -324,32 +313,50 @@ fn append_virtiofs_subcmd_options(app: App<'static, 'static>) -> App<'static, 's
     app.subcommand(subcmd)
 }
 
+fn append_fscache_options(app: App<'static, 'static>) -> App<'static, 'static> {
+    app.arg(
+        Arg::with_name("fscache-tag")
+            .long("fscache-tag")
+            .help("Tag to identify the fscache daemon instance")
+            .takes_value(true)
+            .requires("fscache"),
+    )
+    .arg(
+        Arg::with_name("fscache-threads")
+            .long("fscache-threads")
+            .default_value("4")
+            .help("Number of working threads to serve fscache requests")
+            .takes_value(true)
+            .required(false)
+            .validator(thread_validator),
+    )
+}
+
 fn append_services_subcmd_options(app: App<'static, 'static>) -> App<'static, 'static> {
     let subcmd = SubCommand::with_name("singleton")
-        .about("Run as a global daemon instance to service multiple blobcache/fscache/virtiofs services.")
+        .about(
+            "Run as a global daemon instance to service multiple blobcache/fscache/fuse services.",
+        )
         .arg(
             Arg::with_name("fscache")
                 .long("fscache")
                 .short("F")
                 .help("Working directory for Linux fscache driver to store cached files")
                 .takes_value(true),
-        )
-        .arg(
-            Arg::with_name("fscache-tag")
-                .long("fscache-tag")
-                .help("Tag to identify the fscache daemon instance")
-                .takes_value(true)
-                .requires("fscache"),
-        )
-        .arg(
-            Arg::with_name("fscache-threads")
-                .long("fscache-threads")
-                .default_value("4")
-                .help("Number of working threads to serve fscache requests")
-                .takes_value(true)
-                .required(false)
-                .validator(thread_validator),
         );
+    let subcmd = append_fscache_options(subcmd);
+
+    // TODO: enable support of fuse service
+    /*
+    let subcmd = subcmd.arg(
+        Arg::with_name("fuse")
+            .long("fuse")
+            .short("f")
+            .help("Run as a shared FUSE server"),
+    );
+    let subcmd = append_fuse_options(subcmd);
+    let subcmd = append_fs_options(subcmd);
+    */
```

Comment:
Why do we still need this?

Reply:
The `fuse` and `virtiofs` subcommands are designed for one image per instance mode. And the `singleton` subcommand is designed for one instance for all images mode. So we still need to add support for fuse/virtiofs for singleton.